src/SwapChainManager.cpp

```diff
@@ -0,0 +1,84 @@
+#include "pch.h"
+#include "SwapChainManager.hpp"
+
+namespace D3D12TranslationLayer
+{
+    SwapChainManager::SwapChainManager( D3D12TranslationLayer::ImmediateContext& ImmCtx ) : m_ImmCtx( ImmCtx ) {}
+
+    SwapChainManager::~SwapChainManager()
+    {
+        m_ImmCtx.WaitForCompletion( D3D12TranslationLayer::COMMAND_LIST_TYPE_ALL_MASK );
+    }
+
+    IDXGISwapChain3* SwapChainManager::GetSwapChainForWindow(HWND hwnd, Resource& presentingResource)
+    {
+        auto& spSwapChain = m_SwapChains[hwnd];
+        auto pResourceDesc = presentingResource.AppDesc();
+        DXGI_SWAP_CHAIN_DESC Desc = {};
+        if (spSwapChain)
+        {
+            spSwapChain->GetDesc( &Desc );
+            if (Desc.BufferDesc.Format != pResourceDesc->Format() ||

```

Comment:
You need to swap the format from BGRX -> BGRA earlier, there's 2 more places where the resource's format is used.

Reply:
Doh! Fixed it, thanks. Also noticed that I still had the debug layer flag set for the call to CreateDXGIFactory2, removed that too.